swagger.yaml

```diff
@@ -0,0 +1,350 @@
+        200:
+          description: "A paginated list of import jobs"
+          schema:
+            $ref: '#/definitions/ImportJobStates'
+        400:
+          description: "Request is missing required parameters"
+        500:
+          description: "Failed to process the request due to an internal error"
+  /import/{instanceId}:
+    get:
+      tags:
+      - "External Interface"
+      summary: "Get Information about an import job"
+      description: "Get the current state of an import job, this includes all events which have happened and a overall state of the job."
+      parameters:
+      - $ref: '#/parameters/instanceId'
+      produces:
+      - "application/json"
+      security:
+      - FlorenceAPIKey: []
+      responses:
+        200:
+          description: "Return a single import job"
+          schema:
+            $ref: '#/definitions/ImportJobState'
+        404:
+          description: "InstanceId does not match any import jobs"
+        500:
+          description: "Failed to process the request due to an internal error"
+  /import/{instanceId}/event:
+    put:
+      tags:
+       - "Internal Interface"
+      summary: "Add an event to a import job"
+      description: |
+        Add a new event onto the import job. Events can be for information or error messages. Each event must contain a
+        type of event(Info or Error), a message to explaining the event, time of the event and finally the
+        message offset in kafka
+      parameters:
+      - $ref: '#/parameters/instanceId'
+      - $ref: '#/parameters/event'
+      security:
+      - InternalAPIKey: []
+      responses:
+        200:
+          description: "The event was added to the import job"
+        400:
+          description: "Invalid json message sent"
+        404:
+          description: "InstanceId does not match any import jobs"
+        500:
+          description: "Failed to process the request due to an internal error"
+  /import/{instanceId}/state:
+    put:
+      tags:
+      - "Internal Interface"
+      summary: "Update the state of the import job"
+      description: |
+        Update the state of the import job. The following states are valid.
+        * Created - The job has been created (Default state when created)
+        * Inprogress - The dataset is being converted and ingested into a store
+        * Completed - The dataset has been completed
+        * Failed - The dataset has failed to be completed. See the events messages
+      parameters:
+      - $ref: '#/parameters/instanceId'
+      - $ref: '#/parameters/state'
+      produces:
+      - "application/json"
+      security:
+      - InternalAPIKey: []
+      responses:
+        200:
+          description: "The import state has been updated"
+        404:
+          description: "InstanceId does not match any import jobs"
+        500:
+          description: "Failed to process the request due to an internal error"
+  /import/{instanceId}/dimensions:
```

Comment:
I am wondering whether the URL should specify the dimension value, i.e. 
        /import/{instanceId}/dimensions/{dimension}/options/{option value}